ocp_resources/network_attachment_definition.py

```diff
@@ -9,19 +11,147 @@ class NetworkAttachmentDefinition(NamespacedResource):
     api_group = NamespacedResource.ApiGroup.K8S_CNI_CNCF_IO
     resource_name = None
 
-    def wait_for_status(
-        self, status, timeout=None, label_selector=None, resource_version=None
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        bridge_name=None,
+        cni_type=None,
+        vlan=None,
+        client=None,
+        mtu=None,
+        macspoofchk=None,
+        teardown=True,
+        old_nad_format=False,
+        add_resource_name=True,
+        dry_run=None,
     ):
-        raise NotImplementedError(f"{self.kind} does not have status")
+        super().__init__(
+            name=name,
+            namespace=namespace,
+            client=client,
+            teardown=teardown,
+            dry_run=dry_run,
+        )
+        self.old_nad_format = old_nad_format
+        self.bridge_name = bridge_name
+        self.cni_type = cni_type
+        self.vlan = vlan
+        self.mtu = mtu
+        self.macspoofchk = macspoofchk
+        self.add_resource_name = add_resource_name
 
     def to_dict(self):
         res = super().to_dict()
         if self.yaml_file:
             return res
 
+        res["spec"] = {}
         if self.resource_name is not None:
             res["metadata"]["annotations"] = {
                 f"{NamespacedResource.ApiGroup.K8S_V1_CNI_CNCF_IO}/resourceName": self.resource_name
             }
-        res["spec"] = {}
+
+        spec_config = {"cniVersion": "0.3.1", "name": self.bridge_name}
+        bridge_dict = {"type": self.cni_type, "bridge": self.bridge_name}
```

Comment:
This seems to enforce a key named `bridge`. But not all NAD/s include it.
A generic object should probably only expose generic fields, e.g. the `config`.

Reply:
@EdDev Please review again